File: nodejs/sample/sentenceSplitUsage.js
const {initialize} = require("koalanlp/Util");
const {SentenceSplitter} = require("koalanlp/proc");
const {OKT} = require("koalanlp/API");
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let splitter;

async function processText(text){
    if (text == null || text.trim().length === 0) {
        rl.close();
        return;
    }

    let sentences = await splitter(text);
    sentences.forEach((sent, i) => {
        console.log(`[${i}] ${sent}`);
    });

    readInput();
}

function readInput(){
    rl.question("분석할 문장을 입력하세요>> ", processText);
}

/***********************
 * Main Execution Part *
 ***********************/
initialize({packages: {OKT: 'LATEST'}})  //HNN
    .then(() => {
        splitter = new SentenceSplitter(OKT);
        readInput();
    }).catch((err) => console.error('Error occurred!', err));

File: python/sample/sentence_split_usage.py
from koalanlp.Util import initialize, finalize
from koalanlp.proc import SentenceSplitter
from koalanlp import API

initialize(OKT='LATEST')  #: HNN=2.0.3

splitter = SentenceSplitter(API.OKT)

while True:
    text = input("분석할 문장을 입력하세요>> ").strip()

    if len(text) == 0:
        break

    sentences = splitter(text)

    print("===== Sentence =====")
    for i, sent in enumerate(sentences):
        print("[%s] %s" % (i, sent))

finalize()